src/clj_rn/main.clj

```diff
@@ -67,57 +97,98 @@
                                        :host-ip         host-ip
                                        :js-modules      js-modules
                                        :resource-dirs   resource-dirs
                                        :figwheel-bridge figwheel-bridge})
       (when (= build-id :ios)
         (core/update-ios-rct-web-socket-executor host-ip)
-        (utils/log "Host in RCTWebSocketExecutor.m was updated"))
-      (utils/log (format "Dev server host for %s: %s" platform-name host-ip)))))
+        (utils/println-colorized
+         "Host in RCTWebSocketExecutor.m was updated"
+         shell/color-green))
+      (utils/println-colorized
+       (format "Dev server host for %s: %s" platform-name host-ip)
+       shell/color-green))))
+
+(defmethod task :rebuild-index [[_ & args]]
+  (let [{:keys [build-ids
+                android-device
+                ios-device
+                figwheel-port]}   (parse-cli-options args rebuild-index-task-options)
+        hosts-map                 {:android (core/resolve-dev-host :android android-device)
+                                   :ios     (core/resolve-dev-host :ios ios-device)}]
+    (core/write-env-dev hosts-map figwheel-port)
+    (rebuild-index-files build-ids hosts-map)))
+
+;;; :watch task
+
+(def watch-task-options
+  (with-common-options
+    [[nil "--[no-]start-app" "Start `react-native run-*` or not" :default true]
+     [nil "--[no-]start-figwheel" "Start Figwheel or not" :default true]]))
+
+(defn execute-react-native-cli
+  ([command] (execute-react-native-cli command false))
+  ([command async?]
+   (let [full-command (str "node " react-native-cli " " command)]
+     (if async?
+       (future (shell/exec full-command))
+       (shell/exec full-command)))))
+
+(defn run-builds
+  [build-ids]
+  (doseq [build-id build-ids]
+    (execute-react-native-cli (str "run-" (name build-id)))))
+
+(defn check-react-native-instalation []
+  (when-not (.exists (io/file react-native-cli))
+    (utils/println-colorized "react-native is not found. Please install it first." shell/color-red)
+    (System/exit 1)))
+
+(defmethod task :watch [[_ & args]]
+  (let [{:keys [build-ids
+                android-device
+                ios-device
+                start-figwheel
+                start-app]} (parse-cli-options args watch-task-options)
+        {:keys [figwheel-options builds]} (get-main-config)
+        hosts-map {:android (core/resolve-dev-host :android android-device)
+                   :ios     (core/resolve-dev-host :ios ios-device)}]
+    (check-react-native-instalation)
+    (core/enable-source-maps)
+    (core/write-env-dev hosts-map)
+    (rebuild-index-files build-ids hosts-map)
+    (execute-react-native-cli "start" true)
+    (when start-figwheel
+      (ra/start-figwheel!
+       {:build-ids build-ids
+        :all-builds builds
+        :figwheel-options figwheel-options}))
+    (when start-app (run-builds build-ids))
```

Comment:
As running `watch` task right now, it does so many things (some of them could take a while) that it's hard to follow on which step i'm currently on and what's happening at the moment. Maybe we could print step name before/after each task like:
```
Starting the app..
...
Starting cljs repl...
...
```
in different font & color.

Reply:
`react-native run-*` takes most of the space. Without it output looks more distinct:
```
Patching file node_modules/metro/src/Server/index.js to serve *.map files.
Source maps enabled.
index.ios.js was regenerated
Host in RCTWebSocketExecutor.m was updated
Dev server host for iOS: localhost
Figwheel: Starting server at http://0.0.0.0:3449
Figwheel: Watching build - ios
Figwheel: Cleaning build - ios
Compiling build :ios to "target/ios/app.js" from ["src" "env/dev"]...
Scanning folders for symlinks in /Users/endenwer/projects/clj-rn-example/node_modules (19ms)
┌──────────────────────────────────────────────────────────────────────────────┐
│                                                                              │
│  Running Metro Bundler on port 8081.                                         │
│                                                                              │
│  Keep Metro running while developing on any JS projects. Feel free to        │
│  close this tab and run your own Metro instance if you prefer.               │
│                                                                              │
│  https://github.com/facebook/react-native                                    │
│                                                                              │
└──────────────────────────────────────────────────────────────────────────────┘

Looking for JS files in
   /Users/endenwer/projects/clj-rn-example


Metro Bundler ready.

Loading dependency graph, done.
Successfully compiled build :ios to "target/ios/app.js" in 30.254 seconds.
Figwheel: Starting nREPL server on port: 7888
WARNING: boolean? already refers to: #'clojure.core/boolean? in namespace: mranderson047.toolsanalyzerjvm.v0v6v9.toolsanalyzer.v0v6v7.clojure.tools.analyzer.utils, being replaced by: #'mranderson047.toolsanalyzerjvm.v0v6v9.toolsanalyzer.v0v6v7.clojure.tools.analyzer.utils/boolean?
WARNING: boolean? already refers to: #'clojure.core/boolean? in namespace: mranderson047.toolsanalyzerjvm.v0v6v9.toolsanalyzer.v0v6v7.clojure.tools.analyzer, being replaced by: #'mranderson047.toolsanalyzerjvm.v0v6v9.toolsanalyzer.v0v6v7.clojure.tools.analyzer.utils/boolean?
Launching ClojureScript REPL for build: ios
Figwheel Controls:
          (stop-autobuild)                ;; stops Figwheel autobuilder
          (start-autobuild id ...)        ;; starts autobuilder focused on optional ids
          (switch-to-build id ...)        ;; switches autobuilder to different build
          (reset-autobuild)               ;; stops, cleans, and starts autobuilder
          (reload-config)                 ;; reloads build config and resets autobuild
          (build-once id ...)             ;; builds source one time
          (clean-builds id ..)            ;; deletes compiled cljs target files
          (print-config id ...)           ;; prints out build configurations
          (fig-status)                    ;; displays current state of system
          (figwheel.client/set-autoload false)    ;; will turn autoloading off
          (figwheel.client/set-repl-pprint false) ;; will turn pretty printing off
  Switch REPL build focus:
          :cljs/quit                      ;; allows you to switch REPL to another build
    Docs: (doc function-name-here)
    Exit: :cljs/quit
 Results: Stored in vars *1, *2, *3, *e holds last exception object
Prompt will show when Figwheel connects to your application
```
react native packager runs in parallel and figwheel already has good output. I thought to move `react-native run-*` output to log file and if it fails just tell user to see the log in the file.